src/module.cpp

```diff
@@ -533,11 +533,129 @@ void Module::AddGlobalVariable(const std::string &name, const Type *type, Expr *
             initExpr = TypeCheck(initExpr);
             if (initExpr != nullptr) {
                 // We need to make sure the initializer expression is
-                // the same type as the global.  (But not if it's an
-                // ExprList; they don't have types per se / can't type
-                // convert themselves anyway.)
-                if (llvm::dyn_cast<ExprList>(initExpr) == nullptr)
+                // the same type as the global.
+                if (llvm::dyn_cast<ExprList>(initExpr) == nullptr) {
                     initExpr = TypeConvertExpr(initExpr, type, "initializer");
+                } else {
+                    // The alternative is to create initializer expression with
+                    // correct type and value from ExprList.
+                    // If we have exprList that initalizes const varying int, e.g.:
+                    // static const int x = { 0, 1, 2, 3 };
+                    // then we need to convert initExpr to constExpr that can
+                    // be utilized later in arithmetic expressions that can be
+                    // calculated in compile time (Optimize function call below),
+                    ExprList *exprList = llvm::dyn_cast<ExprList>(initExpr);
+                    if (type->IsConstType() && type->IsVaryingType() && type->IsAtomicType()) {
+                        const AtomicType::BasicType basicType = CastType<AtomicType>(type)->basicType;
+                        const int N = g->target->getVectorWidth();
+                        bool *vals_bool = (bool *)alloca(N * sizeof(bool));
```

Comment:
What's the problem to create vector of `bool`s? As it's not a stack value and there's no deallocation, then there's a memory leak.

Reply:
`alloca` allocates memory on the stack ([man page](https://www.man7.org/linux/man-pages/man3/alloca.3.html)) so no memory leaks is happened. `std::vector<bool>` is discussed in the comment to Arina's change [here](https://github.com/ispc/ispc/pull/2739#discussion_r1471150928).